File: app/Models/VehicleDirection.php
<?php

namespace Irail\Models;

class VehicleDirection
{
    private ?Station $station;
    private string $name;

    public function __construct(string $name, ?Station $station)
    {
        $this->name = $name;
        $this->station = $station;
    }

    /**
     * @return Station|null
     */
    public function getStation(): ?Station
    {
        return $this->station;
    }

    /**
     * @param Station|null $station
     */
    public function setStation(?Station $station): void
    {
        $this->station = $station;
    }

    /**
     * @return string
     */
    public function getName(): string
    {
        return $this->name;
    }

    /**
     * @param string $name
     */
    public function setName(string $name): void
    {
        $this->name = $name;
    }


}

File: src/main/java/be/irail/api/dto/VehicleDirection.java
package be.irail.api.dto;

public class VehicleDirection {
    private StationDto station;
    private String name;

    public VehicleDirection(String name, StationDto station) {
        this.name = name;
        this.station = station;
    }

    /**
     * @return Station
     */
    public StationDto getStation() {
        return this.station;
    }

    /**
     * @param station
     */
    public void setStation(StationDto station) {
        this.station = station;
    }

    /**
     * @return String
     */
    public String getName() {
        return this.name;
    }

    /**
     * @param name
     */
    public void setName(String name) {
        this.name = name;
    }
}
